i#1312 AVX-512 support: Add valign*, vblend*, vcompress*, vexpand* and further opcodes.

hgreving2304 opened the pull request:
Adds the new AVX-512 opcodes valignd, valignq, vblendmps, vblendmpd, vcompressps,
vcompresspd, vexpandps, vexpandpd, vfixupimmps, vfixupimmpd, vfixupimmss, vfixupimmsd,
vgetexppd, vgetexpps, vgetexpsd, vgetexpss, vgetmantpd, vgetmantps, vgetmantsd,
vgetmantss, vpblendmb, vpblendmw, vpblendmd, vpblendmq, vpcompressq, vpcompressd,
vpexpandd, vpexpandq, vptestmb, vptestmw, vptestmd, vptestmq, vptestnmb, vptestnmw,
vptestnmd, vptestnmq.

Adds tests for above.

Opcodes have been checked against llvm-mc, binutils/gas/objdump and capstone.

Issue: #1312

hgreving2304 commented:
run arm tests

derekbruening commented:
> No information found on vpterlogd, vperlogq.

I don't understand this: how are you aware of these if there is really no information at all?  Please elaborate on where these opcodes are listed.  What do they have to do with this PR?  Wouldn't this be information for the issue tracker?

hgreving2304 commented:
run arm tests

hgreving2304 commented:
> > No information found on vpterlogd, vperlogq.
> 
> I don't understand this: how are you aware of these if there is really no information at all? Please elaborate on where these opcodes are listed. What do they have to do with this PR? Wouldn't this be information for the issue tracker?

Yes it doesn't. I had added this comment as I was processing the list in https://docs.google.com/document/d/1gDOL3mCoUCQlCyIg2iuVzxTr5ByVihhg98qD5AuXRVg/edit#.

The instructions is mentioned in a list of new instructions in 5.19 INTEL® ADVANCED VECTOR EXTENSIONS 512 (INTEL® AVX-512), but no other mention of it.

I've removed the comment in the commit message.

hgreving2304 commented:
Looks like spelling error in the Intel manuals. There is vpternlogd/q.
